### ISSUE TITLE
Snap Area doesn't work since update to Version 3.0.9 (150)

### ISSUE THREAD

**Issue by FL0R1AN84** [rxhanson/RectanglePro-Community#353]
Since I installed the update to version 3.0.9 (150) today, the Snap Area is no longer displayed to me. As a test, I deactivated and reactivated this and changed the location of Snap Panel.

My Settings:

![Untitled](https://github.com/rxhanson/RectanglePro-Community/assets/58110115/557f200c-8a84-4efa-a293-eb4decd73325)

My System:

MacBook Pro
Apple M2
macOS 13.5.2

**Comment by rxhanson** [comment]
Thanks for letting me know. Would you mind seeing if we get any logs for when this happens? Hold the option key in the menu bar menu and select View Logging.

Also, if you click and drag more slowly does it work?

**Comment by rxhanson** [comment]
One more thing that might help is if you attach your config so I can load it in and replicate your setup. You can export it from the settings tab of the preferences window. Thanks!

**Comment by FL0R1AN84** [comment]
That's all I got from the log. When I move the window, nothing changes in the log:

```
2023-09-18T20:03:15+02:00: Rectangle Pro v3.0.9 (150)
macOS: Version 13.5.2 (Build 22G91)
Model: Macmini9,1 [3]
License activation: konten@gammagate.de (2022-03-24 20:50:19 +0000)
```

**Comment by FL0R1AN84** [comment]
[RectangleProConfig.json.zip](https://github.com/rxhanson/RectanglePro-My configuaration: Community/files/12651958/RectangleProConfig.json.zip)

**Comment by rxhanson** [comment]
Thanks! Can you try v3.0.10 (beta) and let me know if it fixes the issue?

https://rectangleapp.com/pro/downloads/Rectangle%20Pro%203.0.10.dmg

**Comment by FL0R1AN84** [comment]
Unfortunately not, log also remains empty

`2023-09-19T12:39:38+02:00: Rectangle Pro v3.0.10 (151)
macOS: Version 13.5.2 (Build 22G91)
Model: Mac14,7 [3]
License activation: xxx@xxx.xx (2023-01-02 08:12:06 +0000)`

**Comment by FL0R1AN84** [comment]
@BrMBr Yeah! 

I can confirm that when I close the app "Quick Shade" everything works fine again.

```
2023-09-19T22:03:10+02:00: Rectangle Pro v3.0.9 (150)
macOS: Version 13.5.2 (Build 22G91)
Model: Macmini9,1 [3]
License activation: xxx@xxx.xx (2022-03-24 20:50:19 +0000)
2023-09-19T22:03:16+02:00: AX sizing proposed: (2293.0, 1415.0), result: (2293.0, 1415.0)
2023-09-19T22:03:16+02:00: AX position proposed: (0.0, 25.0), result: (0.0, 25.0)
2023-09-19T22:03:16+02:00: AX sizing proposed: (2293.0, 1415.0), result: (2293.0, 1415.0)
2023-09-19T22:03:16+02:00: firstTwoThirds | display: (0.0, 0.0, 3440.0, 1415.0), calculatedRect: (0.0, 25.0, 2293.0, 1415.0), resultRect: (0.0, 25.0, 2293.0, 1415.0), srcScreen: LG HDR WQHD, destScreen: LG HDR WQHD, resultScreen: LG HDR WQHD
2023-09-19T22:03:17+02:00: AX position proposed: (57.0, 79.0), result: (57.0, 79.0)
2023-09-19T22:03:17+02:00: AX sizing proposed: (2293.0, 1415.0), result: (2293.0, 1415.0)
2023-09-19T22:03:19+02:00: AX sizing proposed: (3440.0, 1415.0), result: (3369.0, 1415.0)
2023-09-19T22:03:19+02:00: AX position proposed: (0.0, 25.0), result: (0.0, 25.0)
2023-09-19T22:03:19+02:00: AX sizing proposed: (3440.0, 1415.0), result: (3440.0, 1415.0)
2023-09-19T22:03:19+02:00: maximize | display: (0.0, 0.0, 3440.0, 1415.0), calculatedRect: (0.0, 25.0, 3440.0, 1415.0), resultRect: (-0.0, 25.0, 3440.0, 1415.0), srcScreen: LG HDR WQHD, destScreen: LG HDR WQHD, resultScreen: LG HDR WQHD
2023-09-19T22:03:22+02:00: AX sizing proposed: (2293.3333333333335, 1415.0), result: (2293.0, 1415.0)
2023-09-19T22:03:22+02:00: AX position proposed: (573.0, 25.0), result: (573.0, 25.0)
2023-09-19T22:03:22+02:00: AX sizing proposed: (2293.3333333333335, 1415.0), result: (2293.0, 1415.0)
2023-09-19T22:03:22+02:00: centerTwoThirds | display: (0.0, 0.0, 3440.0, 1415.0), calculatedRect: (573.0, 25.0, 2293.3333333333335, 1415.0), resultRect: (573.0, 25.0, 2293.0, 1415.0), srcScreen: LG HDR WQHD, destScreen: LG HDR WQHD, resultScreen: LG HDR WQHD
```

**Comment by rxhanson** [comment]
Thanks! The bug fix that went into v3.0.9 was a workaround for a macOS bug that caused glitchy behavior in table views in other parts of macOS and other table views. I'll create a release where you can use a terminal command to switch between these two implementations, and I'll post back when it's ready.

**Comment by ErHaWeb** [comment]
@BrMBr thanks for pointing this out. This way I could find out in my case that the app MonitorControl https://github.com/MonitorControl/MonitorControl is also responsible for the problems. As soon as I quit MonitorControl the snapping works as usual.

**Comment by tilliwilli** [comment]
Just wanted to add here that in my case MacGesture (https://github.com/MacGesture/MacGesture) is also causing the issue to appear. MonitorControl on the other hand causes no problems for me.

**Comment by rxhanson** [comment]
Anyone up for trying out another beta? 

https://rectangleapp.com/pro/downloads/Rectangle%20Pro%203.0.11.dmg

**Comment by FL0R1AN84** [comment]
Yes, that version works for me. I've tested it with [Quick Shade](https://apps.apple.com/us/app/quickshade/id931571202)

**Comment by FL0R1AN84** [comment]
Unfortunately, I still have the problem, whether with MacOS 13.5.2 or with the update to 13.6 - as well as whether with Rectangle Pro 3.0.9, 3.0.10 or 3.0.11 - the snap areas often don't work after a restart. Only restarting or shutting down again can help, at least on my Macmini it often helped. I haven't installed any other apps like Quickshade at the moment and have deactivated all other apps for testing purposes.

```
2023-09-25T12:24:03+02:00: Rectangle Pro v3.0.11 (152)
macOS: Version 13.6 (Build 22G120)
Model: Mac14,7 [3]
License activation: xx@xx.xx (2023-01-02 08:12:06 +0000)
```

Log is empty, too.

**Comment by FL0R1AN84** [comment]
Unfortunately, it still doesn't work on my Mac mini. I don't use MS Teams, etc. - I installed version 3.0.8 again and this version works great.

```
2023-09-25T19:23:28+02:00: Rectangle Pro v3.0.8 (149)
macOS: Version 13.6 (Build 22G120)
Model: Macmini9,1 [3]
```

**Comment by FL0R1AN84** [comment]
It only worked for me for a short time. Even with version 3.0.11 I still have problems on my Mac mini and Mac Pro. At least on my Mac Pro I was able to determine that it was partly due to MS Teams and LanguageTool Desktop. I now use Magnet and have now noticed that both of these apps now work excellently. Unfortunately, not with Rectangle Pro.

At least with version 3.0.8. Everything seems to be going better so far. Although I'm not sure that the problems with MS Teams and Language Tool weren't there before.

**Comment by rxhanson** [comment]
@FL0R1AN84, it's likely that Magnet never accounted for the initial macOS issue that caused the switch in how snap by dragging is handled. In v3.0.11, I included a way to revert back to the old behavior prior to v3.0.9 and it can be enabled via this terminal command:

`defaults write com.knollsoft.Hookshot systemWideMouseDown -int 1`

By reverting back to the old way, you would have the issue where certain tables in macOS and other apps won't respond to clicks properly (only really frustrating if you don't know why it's happening since you can just use your keyboard to change selection).

**Comment by ErHaWeb** [comment]
I can confirm that it now also works with [MonitorControl](https://github.com/MonitorControl/MonitorControl) again.
Thank you!

**Comment by JeanMertz** [comment]
> At least with version 3.0.8. Everything seems to be going better so far. Although, I'm not sure that the issues with MS Teams and LanguageTool weren't there before.

I can confirm that for me, as soon as I quit `LanguageTool`, snap areas started working, without even needing to restart the app. I should also note that the related feature "Snap Panel" also didn't work while the Language Tool app was open.

**Comment by DerGraeflicheGraf** [comment]
> > At least with version 3.0.8. Everything seems to be going better so far. Although, I'm not sure that the issues with MS Teams and LanguageTool weren't there before.
> 
> I can confirm that for me, as soon as I quit `LanguageTool`, snap areas started working, without even needing to restart the app. I should also note that the related feature "Snap Panel" also didn't work while the Language Tool app was open.

Same. It works with BetterDisplay running.

**Comment by tomsch** [comment]
i have the same problem when Keystroke Pro is running.
